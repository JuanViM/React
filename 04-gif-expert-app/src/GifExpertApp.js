import { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = ({defaultCategories = [] }) => {

  //  const categories = ['Dragon ball','One Piece','One puch'];

  const [categories, setCategories] = useState(defaultCategories);  

  //de este modo aÃ±adimos un nuevo elemento al array por el final
//   const handleAdd = () =>{
//       setCategories([...categories,"HunterxHunter"]);
      
//   }

    return (
        <>
        <p>GifExpertApp</p>
        <h2>GifExpertApp</h2>
        <AddCategory setCategories={ setCategories }/>
        <hr />
     

        <ol>
            {
                categories.map(category => (
                     <GifGrid 
                     key={category}
                     category={ category } />
                ))
            }
        </ol>
        </>

    );

}