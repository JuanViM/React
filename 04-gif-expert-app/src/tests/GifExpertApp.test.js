import { GifExpertApp } from "../GifExpertApp";
import "@testing-library/jest-dom";
import {shallow} from "enzyme";



describe('Pruebas con el GifExpertApp', () => { 
    
    test('Prueba del snapshot', () => { 
        
        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();
     });

     test('Debe de mostrar una lista de categorias ', () => { 
         const categories = ["vegeta","one punch"];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>);
        expect( wrapper).toMatchSnapshot();
        expect( wrapper.find("GifGrid").length ).toBe(categories.length);
        
        
      });
     
 })