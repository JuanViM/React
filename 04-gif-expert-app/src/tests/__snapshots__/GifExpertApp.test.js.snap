// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con el GifExpertApp Debe de mostrar una lista de categorias  1`] = `
<Fragment>
  <p>
    GifExpertApp
  </p>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="vegeta"
      key="vegeta"
    />
    <GifGrid
      category="one punch"
      key="one punch"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas con el GifExpertApp Prueba del snapshot 1`] = `
<Fragment>
  <p>
    GifExpertApp
  </p>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
